modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -149,16 +149,18 @@ public void setFlowConstruct(FlowConstruct flowConstruct)
 
     private MetadataContext getMetadataContext() throws MetadataResolvingException
     {
-        //TODO MULE-9530: Improve Config retrieval for Metadata resolution
-        if (!StringUtils.isBlank(configurationProviderName) &&
-            muleContext.getRegistry().get(configurationProviderName) instanceof DynamicConfigurationProvider)
+        MuleEvent fakeEvent = getInitialiserEvent(muleContext);
+        ConfigurationInstance<Object> configuration = getConfiguration(fakeEvent);
```

Comment:
obtain the  provider and if it suits you get a config out of it. Otherwise, you might have initialised a dynamc config which you later discard just because the provider wasn't of the correct ttype